provisioning_client/tests/prov_transport_mqtt_ws_client_ut/prov_transport_mqtt_ws_client_ut.c

```diff
@@ -378,10 +380,10 @@ BEGIN_TEST_SUITE(prov_transport_mqtt_ws_client_ut)
     TEST_FUNCTION(prov_transport_mqtt_ws_register_device_succeed)
     {
         //arrange
-        STRICT_EXPECTED_CALL(prov_transport_common_mqtt_register_device(TEST_DPS_HANDLE, on_transport_json_parse, NULL));
+        STRICT_EXPECTED_CALL(prov_transport_common_mqtt_register_device(TEST_DPS_HANDLE, on_transport_json_parse, on_transport_create_json_payload, NULL));
 
         //act
-        int result = prov_mqtt_ws_transport_register_device(TEST_DPS_HANDLE, on_transport_json_parse, NULL);
+        int result = prov_mqtt_ws_transport_register_device(TEST_DPS_HANDLE, on_transport_json_parse, on_transport_create_json_payload, NULL);
```

Comment:
>on_transport_create_json_payload [](start = 102, length = 32)

A suggestion (which might have some trivial solutions, might not): It appears we are shimming the json payload creation process in every test in this PR. We should have atleast one test that covers obtaining fake ek_value and validating the payload generated by actual production code is passed through to DoWork.

Reply:
MQTT doesn't support TPM.  I'll add that to http and amqp.

---
In reply to: [281253174](https://github.com/Azure/azure-iot-sdk-c/pull/984#discussion_r281253174) [](ancestors = 281253174)